include/rocksdb/options.h

```diff
@@ -1946,6 +1946,20 @@ struct ReadOptions {
   // Default: false
   bool allow_unprepared_value = false;
 
+  // EXPERIMENTAL
+  //
+  // Long-running iterators are holding onto memory and storage resources long
+  // after they are obsolete. This setting (when enabled) will fix that problem
+  // as long as iterators periodically make some progress. The feature is
+  // engineered so that the performance impact should be negligible. We expect
+  // the default value to be true some time in the future.
+  //
+  // Note: Does not have effect on TransactionDB with WRITE_PREPARED or
+  //       WRITE_UNPREPARED policies.
+  //
+  // Default: false
+  bool auto_refresh_iterator_enabled = false;
```

Comment:
Under the newly proposed semantics (private discussion), I suggest `auto_refresh_iterator_with_snapshot` but open to other ideas